src/renderer/utils/cssNames.ts

```diff
@@ -3,29 +3,31 @@
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
 
-// Helper for combining css classes inside components
+import { iter } from "../../common/utils";
 
 export type IgnoredClassNames = number | symbol | Function;
 
 export type IClassName = string | string[] | IClassNameMap | undefined | null | false | IgnoredClassNames;
-export type IClassNameMap = Record<string, unknown>;
+export type IClassNameMap = object;
 
 export function cssNames(...classNames: IClassName[]): string {
-  const classNamesEnabled: IClassNameMap = {};
+  const classNamesEnabled = new Map<string, boolean>();
 
   for (const className of classNames) {
     if (typeof className === "string") {
-      classNamesEnabled[className] = true;
+      classNamesEnabled.set(className, true);
     } else if (Array.isArray(className)) {
       for (const name of className) {
-        classNamesEnabled[name] = true;
+        classNamesEnabled.set(name, true);
       }
     } else if (className && typeof className === "object") {
-      Object.assign(classNamesEnabled, className);
+      for (const [name, value] of Object.entries(className)) {
+        classNamesEnabled.set(name, Boolean(value));
+      }
     }
   }
 
-  return Object.entries(classNamesEnabled)
+  return iter.pipeline(classNamesEnabled)
```

Comment:
Not damning, but still: https://medium.com/bootstart/why-using-chain-is-a-mistake-9bc1f80d51ba

Reply:
Disagree, methods are great for readability and much better for typing. For instance I am trying to emulate https://github.com/tc39/proposal-iterator-helpers.

And to quote yourself back at you, "we control the library for maximum flexibility and control".

Reply:
Resolve when ready :)